### PR TITLE
Remove xfail from torch-cudf.pandas integration test

### PR DESCRIPTION
**Pull request by Matt711**
## Description

The torch test should no longer fail after #16601.

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by Matt711** [comment]
We can merge this PR after #16601

File: python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_pytorch.py [hunk 1]
```diff
@@ -121,8 +121,6 @@ def test_torch_tensor_ctor():
     return torch.tensor(s.values)
 
 
-@pytest.mark.xfail_cudf_pandas(reason="Known failure, see xdf/#210")
-@pytest.mark.xfail_compare
 def test_torch_tensor_from_numpy():
```
Comment (Matt711) [inline python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_pytorch.py:124]: TODO: Check that the test passes

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -304,3 +305,14 @@ jobs:
         node_type: cpu4
         build_type: pull-request
         run_script: "ci/cudf_pandas_scripts/pandas-tests/diff.sh"
+  third-party-integration-tests-cudf-pandas:
```
Comment (Matt711) [inline .github/workflows/pr.yaml:308]: Will remove when CI passes

**Review by jameslamb — APPROVED** [review]
Giving you a `ci-codeowners` approval, expecting a `cudf-python-codeowners` reviewer will be more aware of the potential risk / impact here.

**Review by mroeschke — APPROVED** [review]

**Pull request merged**